lib/rubocop/cop/rails/enum_syntax.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Looks for enums written with keyword arguments syntax.
+      #
+      # Defining enums with keyword arguments syntax is deprecated and will be removed in Rails 8.0.
+      # Positional arguments should be used instead:
+      #
+      # @example
+      #   # bad
+      #   enum status: { active: 0, archived: 1 }, _prefix: true
+      #
+      #   # good
+      #   enum :status, { active: 0, archived: 1 }, prefix: true
+      #
+      class EnumSyntax < Base
+        extend AutoCorrector
+        extend TargetRailsVersion
+
+        minimum_target_rails_version 5.0
```

Comment:
5.0 seems too early. Based on this commit, this syntax was only introduced in rails 7: https://github.com/rails/rails/commit/0618d2d84a501aea93c898aec504ff9a0e09d6f2

Reply:
Good catch! I've updated it!